lib/puppet/type/concat_file.rb

```diff
@@ -58,68 +54,98 @@ def exists?
   end
 
   newparam(:order) do
-    desc "Controls the ordering of fragments. Can be set to alphabetical or numeric."
-    defaultto 'numeric'
+    desc "Controls the ordering of fragments. Can be set to :alpha or :numeric."
+
+    newvalues(:alpha, :numeric)
+
+    defaultto :numeric
   end
 
   newparam(:backup) do
     desc "Controls the filebucketing behavior of the final file and see File type reference for its use."
-    defaultto 'puppet'
+
+    validate do |value|
+      unless [TrueClass, FalseClass, String].include?(value.class)
+        raise ArgumentError, 'Backup must be a Boolean or String'
+      end
+    end
   end
 
-  newparam(:replace) do
+  newparam(:replace, :boolean => true, :parent => Puppet::Parameter::Boolean) do
```

Comment:
:boolean => true just adds a `param?` method to the type, you still need the parent for value checking and coercion.